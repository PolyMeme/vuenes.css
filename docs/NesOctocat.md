# nes-octocat 

Vue Component implementation of the Nes Octocat 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## props 

- `animate` ***Boolean*** (*optional*) 

