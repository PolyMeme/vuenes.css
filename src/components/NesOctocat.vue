<template>
  <li
    :class="{ 'animate': animate }"
    class="nes-octocat"
  />
</template>

<script>
export default {
  props: {
    animate: Boolean
  }
}
</script>
