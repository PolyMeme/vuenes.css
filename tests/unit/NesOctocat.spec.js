import { mount } from '@vue/test-utils'
import NesOctocat from '../../src/components/NesOctocat.vue'

describe('NesOctocat.vue', () => {
  const createOctocatWrapper = (propsData = {}) => {
    return mount(NesOctocat, {
      propsData
    })
  }

  it('should render octocat icon', () => {
    const wrapper = createOctocatWrapper()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.classes()).toContainEqual('octocat')
    expect(wrapper.find('li.octocat').exists()).toBe(true)
  })

  it('should render octocat icon animated of prop animate is set up', () => {
    const wrapper = createOctocatWrapper({
      animate: true
    })

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.classes()).toContainEqual('animate')
    expect(wrapper.find('li.octocat.animate').exists()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = createOctocatWrapper({
      animate: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
