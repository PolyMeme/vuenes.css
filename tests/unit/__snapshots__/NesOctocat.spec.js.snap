// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesOctocat.vue should match snapshot 1`] = `<li class="octocat animate"></li>`;
